go/caddy/pfpki/models.go

```diff
@@ -620,16 +621,31 @@ func (c Cert) download(pfpki *Handler, params map[string]string) (Info, error) {
 	Information := Info{}
 	// Find the Cert
 	var cert Cert
-	if val, ok := params["cn"]; ok {
-		if CertDB := pfpki.DB.Where("Cn = ?", val).Find(&cert); CertDB.Error != nil {
-			Information.Error = CertDB.Error.Error()
-			return Information, errors.New("A database error occured. See log for details.")
+	if profile, ok := params["profile"]; ok {
+		if val, ok := params["cn"]; ok {
+			if CertDB := pfpki.DB.Where("Cn = ? AND profile_id = ?", val, profile).Find(&cert); CertDB.Error != nil {
+				Information.Error = CertDB.Error.Error()
+				return Information, errors.New("A database error occured. See log for details.")
```

Comment:
use a constant for `errors.New("A database error occured. See log for details.")` since its the same error over and over